client/my-sites/site-settings/analytics/form-cloudflare-analytics.js

```diff
@@ -0,0 +1,286 @@
+		isRequestingSettings || isSavingSettings || ! isCodeValid || ! enableForm;
+
+	useEffect( () => {
+		if ( fields?.cloudflare_analytics?.code ) {
+			setIsCloudflareEnabled( true );
+		}
+	}, [ fields ] );
+
+	const handleFieldChange = ( value, callback = () => {} ) => {
+		const updatedCloudflareFields = Object.assign( {}, fields.cloudflare || {}, {
+			code: value,
+		} );
+		updateFields( { cloudflare_analytics: updatedCloudflareFields }, callback );
+	};
+
+	const recordSupportLinkClick = () => {
+		trackTracksEvent( 'calypso_traffic_settings_cloudflare_support_click' );
+	};
+
+	const handleCodeChange = ( event ) => {
+		const code = event.target.value.trim();
+		setIsCodeValid( validateTrackingId( code ) );
+		handleFieldChange( code );
+	};
+
+	const handleFieldFocus = () => {
+		trackTracksEvent( 'calypso_cloudflare_analytics_key_field_focused', { path } );
+		eventTracker( 'Focused Cloudflare Analytics Key Field' )();
+	};
+
+	const handleFieldKeypress = () => {
+		if ( ! loggedCloudflareAnalyticsModified ) {
+			trackTracksEvent( 'calypso_cloudflare_analytics_key_field_modified', { path } );
+			setLoggedCloudflareAnalyticsModified( true );
+		}
+		uniqueEventTracker( 'Typed In Analytics Key Field' )();
+	};
+
+	const processCloudflareCode = () => {
+		const token = fields.cloudflare_analytics?.code?.match( /(?<=token":\s").*?(?=")/ );
+		if ( token ) {
+			handleFieldChange( token );
+		}
+	};
+
+	const handleModalClose = () => {
+		setIsModalVisible( false );
+	};
+	const handleModalConfirm = () => {
+		setIsModalVisible( false );
+		updateFields( { wga: { code: '' } }, () => {
+			handleSubmitForm();
+		} );
+	};
+
+	const handleSaveButtonClick = () => {
+		// display confirmation modal that we will disable GA if Cloudflare is enabled.
+		if ( ! fields?.wga?.code ) {
+			handleSubmitForm();
+			return;
+		}
+		setIsModalVisible( true );
+	};
+
+	const handleFormToggle = () => {
+		if ( isCloudflareEnabled ) {
+			setIsCloudflareEnabled( false );
+			handleFieldChange( '', () => {
+				handleSubmitForm();
+			} );
+		} else {
+			setIsCloudflareEnabled( true );
+		}
+	};
+
+	const renderForm = () => {
+		const placeholderText = isRequestingSettings ? translate( 'Loading' ) : '';
+
+		const nudgeTitle = translate( 'Available with Premium plans or higher' );
+
+		const plan = findFirstSimilarPlanKey( site.plan.product_slug, {
+			type: TYPE_PREMIUM,
+		} );
+
+		const nudge = (
+			<UpsellNudge
+				description={ translate(
+					'Choose an additional analytics tool to connect and get unique insights about your site traffic.'
+				) }
+				event={ 'cloudflare_analytics_settings' }
+				feature={ FEATURE_CLOUDFLARE_ANALYTICS }
+				plan={ plan }
+				href={ null }
+				showIcon={ true }
+				title={ nudgeTitle }
+			/>
+		);
+		return (
+			<form id="analytics" onSubmit={ handleSubmitForm }>
+				{ isModalVisible && (
+					<AnalyticsToggleConfirmationDialog
+						productToDisable="Google Analytics"
+						productToEnable="Cloudflare"
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Cloudflare CDN' )` *ES Score: 6*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/cloudflare-analytics-save-settings)